### PR TITLE
DAOS-12015 control,bio: Add timeout option to LED identify command

### PR DESCRIPTION
**Pull request by tanabarr**
VMD-enabled SSDs can be identified by quickly blinking an LED on the
device using the command dmg storage led identify. This change adds the
ability  to specify a timeout option on the commandline, after which the
SSD LED will stop blinking. If no --timeout value is specified on the
commandline, the SSD LED will blink indefinitely. As timeout values are
tied independently to each SSD, different SSDs can be set to a blinking
state for different lengths of time if multiple command invocations use
different values for different SSDs.

Required-githooks: true

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by NiuYawei — DISMISSED** [review]
The C changes look good to me.

File: src/bio/bio_device.c [hunk 1]
```diff
@@ -777,13 +779,17 @@ led_device_action(void *ctx, struct spdk_pci_device *pci_device)
 }
 
 static int
-check_faulty_devs(struct bio_xs_context *xs_ctxt, bool *is_faulty, struct spdk_pci_addr pci_addr)
+set_timer_and_check_faulty(struct bio_xs_context *xs_ctxt, struct spdk_pci_addr pci_addr,
```
Comment (kjacque) [inline src/bio/bio_device.c:782]: Seems like a case of a function being expanded to do two totally different things. Based on how it's called, it looks like we do either one or the other, not both. It's good to share what can be shared, but I wonder if there is another way to refactor using helper functions to share instead.

**Review by kjacque — APPROVED** [review]
Overall looks good to me. I have one suggestion but I wouldn't block on it.

**Comment by tanabarr** [comment]
The single CI failure is known OSAOnlineExtend issue.
Manual verification of applying LED blinking state with various timeouts and no timeouts has been performed on wolf-156. (Visual verification of LED activity has not been performed on site as this will be covered in test plan).

File: src/bio/bio_device.c (hunk 1 above)
Comment (tanabarr) [reply src/bio/bio_device.c:782]: This helper function wraps a workflow that is required for 2 activities, the set_timer and check_faulty functions use it to apply those 2 distinct activities that both use this workflow. An alternative would be to supply a function pointer to the workflow function but I think this is overkill in this case.

**Review by kjacque — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]

**Pull request merged**